src/cascadia/TerminalApp/MinMaxCloseControl.cpp

```diff
@@ -0,0 +1,57 @@
+﻿//
+// MinMaxCloseControl.xaml.cpp
+// Implementation of the MinMaxCloseControl class
+//
+
+#include "pch.h"
+
+#include "MinMaxCloseControl.h"
+
+namespace winrt::TerminalApp::implementation
+{
+    MinMaxCloseControl::MinMaxCloseControl(uint64_t hWnd)
+        : _window(reinterpret_cast<HWND>(hWnd))
+    {
+        const winrt::Windows::Foundation::Uri resourceLocator{ L"ms-appx:///MinMaxCloseControl.xaml" };
+        winrt::Windows::UI::Xaml::Application::LoadComponent(*this, resourceLocator, winrt::Windows::UI::Xaml::Controls::Primitives::ComponentResourceLocation::Nested);
+    }
+}
+
+#include "MinMaxCloseControl.g.cpp"
+
+void winrt::TerminalApp::implementation::MinMaxCloseControl::OnMaximize(byte flag)
+{
+    POINT point1 = {};
+    ::GetCursorPos(&point1);
+    const LPARAM lParam = MAKELPARAM(point1.x, point1.y);
+    WINDOWPLACEMENT placement = { sizeof(placement) };
+    ::GetWindowPlacement(_window, &placement);
+    if (placement.showCmd == SW_SHOWNORMAL)
+    {
+        ::PostMessage(_window, WM_SYSCOMMAND, SC_MAXIMIZE | flag, lParam);
+    }
+    else if (placement.showCmd == SW_SHOWMAXIMIZED)
+    {
+        ::PostMessage(_window, WM_SYSCOMMAND, SC_RESTORE | flag, lParam);

```

Comment:
I would like to understand the concern more here.

For WinUI 3.0 we are thinking of adding a full feature set Win32 Window Xaml object for Win32 applications.

Reply:
I guess we were more trying to leave ourselves open to the potential of becoming a pure UWP application, should we be something the platform could support. Since UWP's don't have any access to Win32 API's, we were trying to avoid adding win32 code directly to the TerminalApp project, so that the TerminalApp project could easily become a _real_ UWP app by removing the Win32 layer (the WindowsTerminal project) from it.

I don't really know anything about this WinUI 3.0 proposal to have a real Win32 Window exposed - are there any public docs/specs you could link on the topic?

If we do accept this `MinMaxControl` change, would there be a (hopefully simple) way in the future to remove it and replace it with something that's pure UWP, should we ever want to support that scenario? Would the WinUI 3.0 Window get us that? Or would we have to do some sort of check that we can in fact use a Win32 window, and do something else if we can't?